build/kayak-kernel/build.sh

```diff
@@ -35,12 +35,27 @@ fi
 # the build will still work but will use bits from the running system.
 check_for_prebuilt
 
-# pfexec will always work as a nop for root
-[ "$UID" = 0 ] && PFEXEC=pfexec
-
-# Check if privilege escalation is working
-logmsg -n "-- escalating privileges with $PFEXEC"
-[ "`$PFEXEC id -u`" = "0" ] || logerr "Cannot escalate privileges with $PFEXEC"
+if [ -n "$KAYAK_BUILDSEND" ]; then
+    # This is a special case for running in CI. Privileges will not be
+    # escalated and it is expected that the user has sufficient zfs grants
+    # to be able to manipulate the $KAYAK_BUILDSEND filesystem.
+    # The necessary grants are:
+    #     create,destroy,mount,snapshot,rollback,mountpoint,compression
```

Comment:
if they do not have the necessary grants, will the error messages be meaningful?

Reply:
There is a pretty clear message in each case, such as:

```
cannot create 'rpool/kayak': permission denied
```

and generally they would see a permission denied message, although it isn't obvious which grant is missing.

I only expect these new variables to be set in CI as I want the miniriit build to go through as it's where we often miss new libraries etc. and only find out when we come to build media.